From 173cff7fcdee88cc5da8bc299f42156886873310 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 22 Feb 2024 10:20:38 +0100
Subject: [PATCH 0684/1695] plat-stm32mp2: conf: enable PKA for stm32mp2
 platforms

Default enables stm32_pka and crypto framework for stm32mp2 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a70c8b381cfd61c7ba7774a0a1b9402fdfb0b5c2 plat-stm32mp2: conf: enable PKA for stm32mp2 platforms]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: I96d241ffdcc9b87342616952ab3b57c57b5d936b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362226
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index b168d7910..3c012913b 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -135,6 +135,7 @@ CFG_STM32_IPCC ?= y
 CFG_STM32_IWDG ?= y
 CFG_STM32_LPTIMER ?= y
 CFG_STM32_OMM ?= y
+CFG_STM32_PKA ?= y
 CFG_STM32_PWR_IRQ ?= y
 CFG_STM32_RIF ?= y
 CFG_STM32_RIFSC ?= y
@@ -181,7 +182,7 @@ override CFG_STM32_CRYP := n
 endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
 
 # If any crypto driver is enabled, enable the crypto-framework layer
-ifeq ($(call cfg-one-enabled, CFG_STM32_HASH CFG_STM32_SAES),y)
+ifeq ($(call cfg-one-enabled, CFG_STM32_HASH CFG_STM32_PKA CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-- 
2.47.2

